recruit/tags.go

```diff
@@ -0,0 +1,380 @@
+		return AddTagsResponse{}, fmt.Errorf("failed to insert Tag(s): %s", err)
+	}
+
+	if v, ok := endpoint.ResponseData.(*AddTagsResponse); ok {
+		return *v, nil
+	}
+
+	return AddTagsResponse{}, fmt.Errorf("data returned was not 'AddTagsResponse'")
+}
+
+// https://www.zoho.com/recruit/developer-guide/apiv2/add-tags.html
+func (c *API) AddTagsToId(module Module, ID string, params map[string]zoho.Parameter) (data AddTagsResponse, err error) {
+	endpoint := zoho.Endpoint{
+		Name:         "AddTagsToId",
+		URL:          fmt.Sprintf("https://recruit.zoho.%s/recruit/v2/%s/%s/actions/add_tags", c.ZohoTLD, module, ID),
+		Method:       zoho.HTTPPost,
+		ResponseData: &AddTagsResponse{},
+		URLParameters: map[string]zoho.Parameter{
+			"tag_names": "",
+		},
+	}
+
+	if len(params) > 0 {
+		for k, v := range params {
+			endpoint.URLParameters[k] = v
+		}
+	}
+
+	err = c.Zoho.HTTPRequest(&endpoint)
+	if err != nil {
+		return AddTagsResponse{}, fmt.Errorf("failed to add Tag(s): %s", err)
+	}
+
+	if v, ok := endpoint.ResponseData.(*AddTagsResponse); ok {
+		for _, resp := range v.Data {
+			if resp.Code != "SUCCESS" {
+				return AddTagsResponse{}, fmt.Errorf("failed to add Tag(s): %s: %s", resp.Code, resp.Message)
+			}
+		}
+		return *v, nil
+	}
+
+	return AddTagsResponse{}, fmt.Errorf("data returned was not 'AddTagsResponse'")
+}
+
+type AddTagsResponse struct {
+	Data []struct {
+		Code    string `json:"code"`
+		Details struct {
+			ID   int64    `json:"id"`
+			Tags []string `json:"tags"`
+		} `json:"details"`
+		Message string `json:"message"`
+		Status  string `json:"status"`
+	} `json:"data"`
+}
+
+// https://www.zoho.com/recruit/developer-guide/apiv2/delete-tag.html
+func (c *API) DeleteTagById(tagID string) (data DeleteTagResponse, err error) {
+	if len(tagID) == 0 {
+		return DeleteTagResponse{}, fmt.Errorf("failed to delete Tag, must provide tagID")
+	}
+
+	endpoint := zoho.Endpoint{
+		Name:         "DeleteTagById",
+		URL:          fmt.Sprintf("https://recruit.zoho.%s/recruit/v2/settings/tags/%s", c.ZohoTLD, tagID),
+		Method:       zoho.HTTPDelete,
+		ResponseData: &DeleteTagResponse{},
+	}
+
+	err = c.Zoho.HTTPRequest(&endpoint)
+	if err != nil {
+		return DeleteTagResponse{}, fmt.Errorf("failed to delete Tag: %s", err)
+	}
+
+	if v, ok := endpoint.ResponseData.(*DeleteTagResponse); ok {
+		return *v, nil
+	}
+
+	return DeleteTagResponse{}, fmt.Errorf("data retrieved was not 'DeleteTagResponse'")
+}
+
+type DeleteTagResponse struct {
+	Tags struct {
+		Code    string `json:"code"`
+		Details struct {
+			ID int64 `json:"id"`
+		} `json:"details"`
+		Message string `json:"message"`
+		Status  string `json:"status"`
+	} `json:"tags"`
+}
+
+// https://www.zoho.com/recruit/developer-guide/apiv2/get-tag-list.html
+func (c *API) GetTagsList(params map[string]zoho.Parameter) (data TagsListResponse, err error) {
+	if len(params["module"]) == 0 {
+		return TagsListResponse{}, fmt.Errorf("failed to list Tags, module name is missing")
+	}
```

Comment:
I would instead:
```GetTagsList(module Module)```

```if module == "" {
    return TagsListResponse{}, fmt.Errorf("failed to list Tags, module name missing")
}```

Reply:
i could do this only if you ignore second param, 'my_tags"
https://www.zoho.com/recruit/developer-guide/apiv2/get-tag-list.html

Reply:
You can define the methods type signature like:
```
func (c *API) GetTagsList(module Module, params map[string]zoho.Parameter) (data TagsListResponse, err error)
```

And in the function:
```
endpoint := zoho.Endpoint{
......
		URLParameters: map[string]zoho.Parameter{
			"module":  module, // mandatory
			"my_tags": "",
		},
	}

	if len(params) > 0 {
		for k, v := range params {
			endpoint.URLParameters[k] = v
		}
	}
```